### ISSUE TITLE
Native SQLModel Support

### ISSUE THREAD

**Issue by awtkns** [awtkns/fastapi-crudrouter#109]
### Discussed in https://github.com/awtkns/fastapi-crudrouter/discussions/108

<div type='discussions-op-text'>

<sup>Originally posted by **voice1** September 29, 2021</sup>
Is there any intent to support SQLModel? [https://github.com/tiangolo/sqlmodel](https://github.com/tiangolo/sqlmodel)</div>

**Comment by nuno-andre** [comment]
Hi! FWIW, SQLModel [replicates both SQLAlchemy and Pydantic](https://github.com/tiangolo/sqlmodel/blob/a36c6d5778e23bb310c18992b2f301ffb34e8c8e/sqlmodel/main.py#L212). It seems that supporting it may be very straightforward. This just works:

```python
from typing import Optional
from sqlmodel import Field, SQLModel, create_engine
from sqlalchemy.orm import sessionmaker

from fastapi import FastAPI
from fastapi_crudrouter import SQLAlchemyCRUDRouter

app = FastAPI()
engine = create_engine(
    "sqlite:///./app.db",
    connect_args={"check_same_thread": False}
)

SessionLocal = sessionmaker(
    autocommit=False,
    autoflush=False,
    bind=engine
)

def get_db():
    session = SessionLocal()
    try:
        yield session
        session.commit()
    finally:
        session.close()

class Potato(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    color: str
    mass: float

router = SQLAlchemyCRUDRouter(
    schema=Potato,
    db_model=Potato,
    db=get_db,
    prefix='potato'
)

SQLModel.metadata.create_all(bind=engine)

app.include_router(router)
```

Regards!

**Comment by ChuckMoe** [comment]
> ```python
> class Potato(SQLModel, table=True):
>     # this idiom is no longer needed due to CreatePotato
>     # id: Optional[int] = Field(default=None, primary_key=True)
>     id: int = Field(primary_key=True)
>     color: str
>     mass: float
>     created_at: datetime = Field(default_factory=func.now)
> 
> class CreatePotato(BaseModel):
>     color: str
>     mass: float
> 
> class UpdatePotato(BaseModel):
>     color: str
> ```

You don't even need to use the pydantic BaseModel, you could also just use the SQLModel as it also doubles as pydantic model. This enables us to reuse our classes like so:

```python
class BasePotato(SQLModel):
    color: str

class CreatePotato(BasePotato):
    mass: float

class Potato(CreatePotato, table=True):
    id: int = Field(primary_key=True)
    created_at: datetime = Field(defaukt_factory=func.now)
```

**Comment by nuno-andre** [comment]
Good point, @ChuckMoe! It's truly impressive the amount of boilerplate that can be reduced with SQLModel and FastAPI-CRUDRouter. I've updated the example with the _UpdateModel_ > _CreateModel_ > _BaseModel_ approach.

**Comment by russdot** [comment]
Finding this post made my day!  I've been able to integrate FastAPI with [FastAPI-Users](https://fastapi-users.github.io/fastapi-users/10.2/) (using SQLAlchemy + `asyncio`) and now `fastapi-crudrouter` 👍 

Using the `Potato` schema/models above, I can observe an entire new Potato section in the `/docs`. However, when I attempt to create a new potato via `POST`, I'm seeing the following:

```
pydantic.error_wrappers.ValidationError: 3 validation errors for Potato
response -> id
  none is not an allowed value (type=type_error.none.not_allowed)
response -> created_at
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
response -> updated_at
  invalid type; expected datetime, string, bytes, int or float (type=type_error)
```
I'm still learning Pydantic - but these fields are all defined in the `Potato` model and expected to be filled upon creation? Maybe I'm missing something? Any help is greatly appreciated. This extension is fantastic - thank you!
